test/integration/apiserver/apply/apply_test.go

```diff
@@ -2811,3 +2811,187 @@ spec:
+		AbsPath("/apis/apps/v1").
+		Name("deployment").
+		Resource("deployments").
+		SubResource("scale").
+		Namespace("default").
+		Param("fieldManager", "scale_test").
+		Body([]byte(`{"spec":{"replicas": 5}}`)).
+		Do(context.TODO()).
+		Get()
+	if err != nil {
+		t.Fatalf("Error updating scale subresource: %v ", err)
+	}
+
+	deployment, err = client.AppsV1().Deployments("default").Get(context.TODO(), "deployment", metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Failed to retrieve object: %v", err)
+	}
+
+	if *deployment.Spec.Replicas != 5 {
+		t.Fatalf("Expected replicas to be 5, but got: %d", *deployment.Spec.Replicas)
+	}
+
+	assertCompleteOwnership(t, (*deployment).GetManagedFields(), "scale_test")
+
+	// Re-apply the original object, it should fail with conflict because replicas have changed
+	_, err = client.CoreV1().RESTClient().Patch(types.ApplyPatchType).
+		AbsPath("/apis/apps/v1").
+		Namespace("default").
+		Resource("deployments").
+		Name("deployment").
+		Param("fieldManager", "apply_test").
+		Body(deploymentWithReplicas(1)).
+		Do(context.TODO()).
+		Get()
+	if !apierrors.IsConflict(err) {
+		t.Fatalf("Expected conflict error but got: %v", err)
+	}
+
+	// Re-apply forcing the changes should succeed
+	_, err = client.CoreV1().RESTClient().Patch(types.ApplyPatchType).
+		AbsPath("/apis/apps/v1").
+		Namespace("default").
+		Resource("deployments").
+		Name("deployment").
+		Param("fieldManager", "apply_test").
+		Param("force", "true").
+		Body(deploymentWithReplicas(1)).
+		Do(context.TODO()).
+		Get()
+	if err != nil {
+		t.Fatalf("Error updating deployment: %v ", err)
+	}
+
+	deployment, err = client.AppsV1().Deployments("default").Get(context.TODO(), "deployment", metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Failed to retrieve object: %v", err)
+	}
+
+	if *deployment.Spec.Replicas != 1 {
+		t.Fatalf("Expected replicas to be 1, but got: %d", *deployment.Spec.Replicas)
+	}
+
+	assertCompleteOwnership(t, (*deployment).GetManagedFields(), "apply_test")
+
+	// Replace scale object
+	_, err = client.CoreV1().RESTClient().
+		Put().
+		AbsPath("/apis/apps/v1").
+		Namespace("default").
+		Resource("deployments").
+		SubResource("scale").
+		Name("deployment").
+		Param("fieldManager", "replace_test").
+		Body([]byte(`{"kind":"Scale","apiVersion":"autoscaling/v1","metadata":{"name":"deployment","namespace":"default"},"spec":{"replicas":7}}`)).
+		Do(context.TODO()).
+		Get()
+	if err != nil {
+		t.Fatalf("Error updating deployment: %v ", err)
+	}
+
+	deployment, err = client.AppsV1().Deployments("default").Get(context.TODO(), "deployment", metav1.GetOptions{})
+	if err != nil {
+		t.Fatalf("Failed to retrieve object: %v", err)
+	}
+
+	if *deployment.Spec.Replicas != 7 {
+		t.Fatalf("Expected to replicas to be 7, but got: %d", *deployment.Spec.Replicas)
+	}
+
+	assertCompleteOwnership(t, (*deployment).GetManagedFields(), "replace_test")
+}
+
+func deploymentWithReplicas(replicas int) []byte {
```

Comment:
nit: maybe inline this in the test? It looks like we use the same object several times.

Reply:
I agree, it would be cleaner 👍